integration-tests/smoke/ccip/ccip_reorg_test.go

```diff
@@ -0,0 +1,675 @@
+	e, _, tEnv := testsetups.NewIntegrationEnvironment(
+		t,
+		testhelpers.WithLogMessagesToIgnore([]testhelpers.LogMessageToIgnore{
+			{
+				Msg:    "Got very old block.",
+				Reason: "We are expecting a re-org beyond finality",
+				Level:  zapcore.DPanicLevel,
+			},
+			{
+				Msg:    "Reorg greater than finality depth detected",
+				Reason: "We are expecting a re-org beyond finality",
+				Level:  zapcore.DPanicLevel,
+			},
+			{
+				Msg:    "Failed to poll and save logs due to finality violation, retrying later",
+				Reason: "We are expecting a re-org beyond finality",
+				Level:  zapcore.DPanicLevel,
+			},
+		}),
+	)
+
+	nodeInfos, err := deployment.NodeInfo(e.Env.NodeIDs, e.Env.Offchain)
+	require.NoError(t, err)
+
+	var nonBootstrapP2PIDs = make([]string, 0, len(nodeInfos.NonBootstraps()))
+	for _, n := range nodeInfos.NonBootstraps() {
+		nonBootstrapP2PIDs = append(nonBootstrapP2PIDs, strings.TrimPrefix(n.PeerID.String(), "p2p_"))
+	}
+
+	l.Info().Msgf("nonBootstrapP2PIDs: %s", nonBootstrapP2PIDs)
+
+	state, err := ccipcs.LoadOnchainState(e.Env)
+	require.NoError(t, err)
+
+	allChains := e.Env.AllChainSelectors()
+	require.GreaterOrEqual(t, len(allChains), 2)
+	sourceSelector := allChains[0]
+	destChainSelector := allChains[1]
+	destChain, ok := chainsel.ChainBySelector(destChainSelector)
+	require.True(t, ok)
+	destLogPollerService := fmt.Sprintf("EVM.%d.LogPoller", destChain.EvmChainID)
+	l.
+		Info().
+		Msgf("reorging dest log poller service name: %s", destLogPollerService)
+
+	dockerEnv, ok := tEnv.(*testsetups.DeployedLocalDevEnvironment)
+	require.True(t, ok)
+
+	chainSelToRPCURL := make(map[uint64]string)
+	for _, chain := range dockerEnv.GetDevEnvConfig().Chains {
+		require.GreaterOrEqual(t, len(chain.HTTPRPCs), 1)
+		details, err := chainsel.GetChainDetailsByChainIDAndFamily(strconv.FormatUint(chain.ChainID, 10), chainsel.FamilyEVM)
+		require.NoError(t, err)
+
+		chainSelToRPCURL[details.ChainSelector] = chain.HTTPRPCs[0].Internal
+	}
+
+	reorgDestClient := ctf_client.NewRPCClient(chainSelToRPCURL[destChainSelector], nil)
+
+	// setup lanes from s1 and s2 to destChainSelector
+	testhelpers.AddLaneWithDefaultPricesAndFeeQuoterConfig(t, &e, state, sourceSelector, destChainSelector, false)
+
+	// wait for log poller filters to get registered.
+	l.Info().Msg("waiting for log poller filters to get registered")
+	time.Sleep(15 * time.Second)
+	reorgingMsgEvent := testhelpers.TestSendRequest(t, e.Env, state, sourceSelector, destChainSelector, false, router.ClientEVM2AnyMessage{
+		Receiver:     common.LeftPadBytes(state.Chains[destChainSelector].Receiver.Address().Bytes(), 32),
+		Data:         []byte("hello world"),
+		TokenAmounts: nil,
+		FeeToken:     common.HexToAddress("0x0"),
+		ExtraArgs:    nil,
+	})
+	l.Info().Msgf("sent CCIP message that will get re-orged, msg id: %x", reorgingMsgEvent.Message.Header.MessageId)
+
+	// Run reorg above finality depth
+	const reorgDepth = 60
```

Comment:
would be easier to update if you put something like these into var 
`ReorgDepthGreaterThanFinality` and `ReorgDepthLesserThanFinality`